src/server/api/consume-wifi-data.ts

```diff
@@ -0,0 +1,109 @@
+import { Kafka } from "kafkajs";
+import { SchemaRegistry } from "@kafkajs/confluent-schema-registry";
+import { serverSupabaseClient } from "#supabase/server";
+import type { SupabaseClient } from '@supabase/supabase-js';
+
+const { internalSecret, kafkaConfig, schemaRegistry } = useRuntimeConfig();
+
+console.time("Initialize");
+const kafka = new Kafka({
+  clientId: "tudelft-spacefinder",
+  ssl: true,
+  sasl: {
+    mechanism: "plain",
+    username: kafkaConfig.username,
+    password: kafkaConfig.password,
+  },
+  brokers: [kafkaConfig.url],
+});
+
+const registry = new SchemaRegistry({
+  host: schemaRegistry.url,
+  auth: {
+    username: schemaRegistry.username,
+    password: schemaRegistry.password,
+  },
+});
+
+const consumer = kafka.consumer({ groupId: kafkaConfig.groupId });
+console.timeEnd("Initialize");
+
+let seeked = false;
+
+export default defineEventHandler(async (event) => {
```

Comment:
If I imagine I want to unit-test it, this function looks a bit like it does too many different things - being a server route handler, connecting to and consuming from Kafka, transforming its data, upserting to Supabase. If we have time I would suggest to refactor out smaller functions with more limited responsibilities, maybe as a separate PR? Maybe even actually write a test to illustrate how we transform decoded data into what we insert in the table?

Reply:
Yes sounds good! A new PR for refactoring would be nice so this is no longer in the air. I thought about this and think the only thing we can actually test is the data transformation after it is decoded.

Reply:
Yeah, and it's probably the only thing that has sense to test. I guess this kind of test could serve as documentation